src/Simplify.cpp

```diff
@@ -5591,6 +5611,14 @@ void check_boolean() {
                            Evaluate::make(x+foo_simple+1),
                            Evaluate::make(x+19)));
 
+    // The construct
+    //     if (var == expr) then a else b;
+    // was being simplified incorrectly, but *only* if var was of type Bool.
+    Stmt then_clause = AssertStmt::make(b2, Expr(22));
+    Stmt else_clause = AssertStmt::make(b2, Expr(33));
+    check(IfThenElse::make(b1 == true, then_clause, else_clause),
```

Comment:
Good point. Changed to b1 == b2 (and verified that this still triggered the issue if the fix is backed out).